### PR TITLE
Add support for blinding pegins

### PR DESCRIPTION
**Pull request by stevenroose**
Adds pegin handling support for `blindrawtransaction` and `rawblindrawtransaction`.

### CODE REVIEW

File: test/functional/feature_fedpeg.py [hunk 1]
```diff
@@ -524,6 +531,63 @@ def run_test(self):
...
+        sidechain.sendtoaddress(blind_addr, 15)
+        sidechain.generate(5)
+        # start pegin
+        pegin_addrs = sidechain.getpeginaddress()
+        assert_equal(sidechain.decodescript(pegin_addrs["claim_script"])["type"], "witness_v0_keyhash")
+        pegin_addr = addrs["mainchain_address"]
+        txid_fund = parent.sendtoaddress(pegin_addr, 10)
+        # 10+2 confirms required to get into mempool and confirm
+        parent.generate(11)
+        time.sleep(2)
```
Comment (instagibbs) [inline test/functional/feature_fedpeg.py]: what is sleep for?

File: test/functional/feature_fedpeg.py [hunk 2]
```diff
@@ -565,28 +565,44 @@ def run_test(self):
...
+            assert("value" not in final_decoded["vout"][1])
+            assert("asset" not in final_decoded["vout"][1])
+            assert(final_decoded["vout"][1]["commitmentnonce_fully_valid"])
+            assert("value" in final_decoded["vout"][2])
+            assert("asset" in final_decoded["vout"][2])
+            # check that it is accepted in the mempool
+            accepted = sidechain.testmempoolaccept([pegin_signed["hex"]])[0]
+            if not accepted["allowed"]:
+                print("Blinded pegin not allowed in mempool; reject-reason: %s" % accepted["reject-reason"])
+                raise Exception()
```
Comment (instagibbs) [inline test/functional/feature_fedpeg.py]: just stick the message inside the `Exception`

File: test/functional/feature_fedpeg.py [hunk 3]
```diff
@@ -565,28 +565,44 @@ def run_test(self):
...
+            print("Blinded transaction looks ok!") # need this print to distinguish failures in for loop
+        # check if they get mined; since we're trying to mine two double spends, disconnect the nodes
+        disconnect_nodes(sidechain, 3)
+        disconnect_nodes(sidechain2, 2)
+        txid1 = sidechain.sendrawtransaction(pegin_signed1["hex"])
+        blocks = sidechain.generate(3)
+        assert_equal(sidechain.getrawtransaction(txid1, True, blocks[0])["confirmations"], 3)
+        txid2 = sidechain2.sendrawtransaction(pegin_signed2["hex"])
+        blocks = sidechain2.generate(3)
+        assert_equal(sidechain2.getrawtransaction(txid2, True, blocks[0])["confirmations"], 3)
```
Comment (instagibbs) [inline test/functional/feature_fedpeg.py]: I'd prefer you put the test back in a usable state after this. i.e., re-connect the two nodes at least.

**Comment by instagibbs** [comment]
looks good otherwise, and nice to see the python stuff put to work :P

File: test/functional/feature_fedpeg.py (hunk 1 above)
Comment (stevenroose) [reply test/functional/feature_fedpeg.py]: no idea :D it was done after the parent send in the rest of the test as well. I think it'll work without here, not sure Travis :D

File: test/functional/feature_fedpeg.py (hunk 2 above)
Comment (stevenroose) [reply test/functional/feature_fedpeg.py]: haha, nice!

File: test/functional/feature_fedpeg.py (hunk 3 above)
Comment (stevenroose) [reply test/functional/feature_fedpeg.py]: hmm yeah I was thinking that it was not needed but someone extending this test should do it. I'll connect them. I was looking for a way to discard the tx from the mempool so I could mine the first one, invalidate block, discard and mine the other one, but I couldn't find how.

File: test/functional/feature_fedpeg.py (hunk 1 above)
Comment (stevenroose) [reply test/functional/feature_fedpeg.py]: removed it and pushed, let's see :) it works locally

**Comment by stevenroose** [comment]
> and nice to see the python stuff put to work :P

Yeah nice to see that we can do all that stuff manually. I think the only thing that we can't do without liquid is the blinding, but that should be possible in Rust with rust-secp256k1-zkp. Very excited about that!

**Comment by stevenroose** [comment]
@instagibbs build is fine it seems

**Comment by instagibbs** [comment]
re:"Always return blind address for "blech32" type"

The help for getnewaddress should reflect this.

**Comment by instagibbs** [comment]
utACK https://github.com/ElementsProject/elements/pull/663/commits/cc491065147276193e39095ee0f54b913a37180d

**Pull request merged**